arangod/IResearch/Search.cpp

```diff
@@ -0,0 +1,441 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Valery Mironov
+////////////////////////////////////////////////////////////////////////////////
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Aql/QueryCache.h"
+#include "Basics/Result.h"
+#include "Basics/debugging.h"
+#include "Basics/Exceptions.h"
+#include "Basics/DownCast.h"
+#include "Cluster/ClusterFeature.h"
+#include "Cluster/ClusterInfo.h"
+#include "Indexes/Index.h"
+#include "IResearch/Search.h"
+#include "IResearch/IResearchInvertedIndex.h"
+#include "IResearch/ViewSnapshot.h"
+#include "RestServer/ViewTypesFeature.h"
+#include "Transaction/Methods.h"
+#include "VocBase/LogicalCollection.h"
+#include "Utils/CollectionNameResolver.h"
+#include "Utils/Events.h"
+
+#ifdef USE_PLAN_CACHE
+#include "Aql/PlanCache.h"
+#endif
+
+#include <absl/strings/numbers.h>
+#include <absl/strings/str_cat.h>
+
+namespace arangodb::iresearch {
+namespace {
+
+std::shared_ptr<LogicalCollection> getCollection(
+    CollectionNameResolver& resolver, velocypack::Slice cidOrName) {
+  if (cidOrName.isString()) {
+    return resolver.getCollection(cidOrName.stringView());
+  } else if (cidOrName.isNumber<DataSourceId::BaseType>()) {
+    std::this_thread::sleep_for(std::chrono::seconds{10});
```

Comment:
Is this sleep intentional? If so, what is the reason behind that? Maybe add a comment why to sleep and why exactly 10s ?